cores/esp8266/base64.h

```diff
@@ -1,39 +1,39 @@
-/**
- * base64.h
- *
- * Created on: 09.12.2015
- *
- * Copyright (c) 2015 Markus Sattler. All rights reserved.
- * This file is part of the ESP8266 core for Arduino.
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 2.1 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
- *
- */
-
-#ifndef CORE_BASE64_H_
-#define CORE_BASE64_H_
-
-class base64 {
-    public:
-        // NOTE: The default behaviour of backend (lib64)
-        // is to add a newline every 72 (encoded) characters output.
-        // This may 'break' longer uris and json variables
-        static String encode(const uint8_t * data, size_t length, bool doNewLines = true);
-        static String encode(const String& text, bool doNewLines = true);
-    private:
-};
-
-
-#endif /* CORE_BASE64_H_ */
+/**
+ * base64.h
+ *
+ * Created on: 09.12.2015
+ *
+ * Copyright (c) 2015 Markus Sattler. All rights reserved.
+ * This file is part of the ESP8266 core for Arduino.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef CORE_BASE64_H_
+#define CORE_BASE64_H_
+
+namespace base64 {
```

Comment:
This may break current applications. It is the Arduino way to instantiate objects, even empty ones, before use. Current applications that have something like
```cpp
base64 somename;
...
somename.encode(blah);
```
anywhere in their code won't compile anymore. In contrast, the current implementation allows both instantiated and non-instantiated usage. 

If you feel strongly about this change, we can keep it, but it is a breaking change. In that case, we can target this entire PR for v3, or this breaking part of the PR can be split into a separate PR to be targeted for v3, and we only merge the other performance enhancement now. It's your choice.

Reply:
thanks for the explanation, finally understood the reasoning behind it. reverted this part.